src/shims/unix/fs.rs

```diff
@@ -1659,6 +1664,146 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
             }
         }
     }
+
+    fn mkstemp(&mut self, template_op: &OpTy<'tcx, Tag>) -> InterpResult<'tcx, i32> {
+        use rand::distributions::{Distribution, Uniform};
+        #[cfg(target_family = "unix")]
+        use std::ffi::OsStr;
+        use std::ffi::OsString;
+        #[cfg(target_family = "unix")]
+        use std::os::unix::{ffi::OsStrExt, fs::OpenOptionsExt};
+        #[cfg(target_family = "windows")]
```

Comment:
Also you can't do this kind of check unfortunately. This is checking for the host's target, not the actual target. You need to use something like https://github.com/rust-lang/miri/blob/e837beede026b646079f0ec6351c54717a96a79d/src/shims/foreign_items.rs#L704 for that. But that can't be used for imports ^^ So you need to write platform independent logic handling platform specific things. It's very confusing. It may end up being simpler writing the logic twice

Reply:
Yeah, this code is testing the *host* target. That's very odd and needs justification via comments.

Reply:
We need to check `char`s and generate `OsString`s which are different between host and target.

Reply:
The containing `bytes_to_os_str` is wrong though...windows paths do not have to be utf8 so miri is adding additional restrictions that don't exist. https://github.com/dylni/os_str_bytes looks much more complete...perhaps I will add it as a follow-up.

Reply:
> The containing bytes_to_os_str is wrong though

It will never produce wrong results, but indeed it can fail for non-UTF-8 paths on Windows. It'd be good to support them but the current stdlib APIs won't let us. 

However, https://github.com/dylni/os_str_bytes/blob/master/src/windows seems to do something Wtf8-specific, which I am not convinced makes sense for Miri -- we cannot assume that the stdlib uses Wtf8, and neither does windows.